lib/requestIterator.js

```diff
@@ -28,7 +31,13 @@ RequestIterator.prototype.nextRequestBuffer = function () {
 
 RequestIterator.prototype.move = function () {
   // get the current buffer and proceed to next request
-  const ret = this.currentRequest.requestBuffer
+  let ret = this.currentRequest.requestBuffer
+  if (this.reqDefaults.idReplacement) {
+    const contentLengthRegex = /\[<contentLength>\]/
+    const reqStr = ret.toString().replace(/\[<id>\]/g, hyperid())
+    const bodyLength = reqStr.split(contentLengthRegex)[1].trim().length
+    ret = new Buffer(reqStr.replace(contentLengthRegex, `${bodyLength}`))
```

Comment:
The only problem with moving it into requestBuilder is that the length of the replacement value on line 37 will vary. As you will know, hyperid generates IDs made up of a base64 string plus an integer, the integer will grow in length as it gets larger, which prevents setting content length until all replacements have been carried out.

This would not be a problem if we used UUIDs as they would be a fixed size, of course then there is a penalty for constantly regenerating UUIDs. I could fork and submit a change to the implementation of hyperid to pad the iteration section out to a fixed width but I don't know how expensive that would be in terms of the performance of hyperid. What do you think?